### ISSUE TITLE
Unread messages do not persist over session disconnect

### ISSUE THREAD

**Issue by logan-jobzmall** [pubnub/chat-engine-unread-messages#4]
@ianjennings,

I can see in the code that unread messages are only counted when the user is online and receiving events. Is there any way to see what was unread once the user logs back in?

Logan

**Comment by devbazilio** [comment]
Agreed. It's working only for active connected chats that are listed in ChatEngine.chats

**Comment by nishith-pubnub** [comment]
@logan-jobzmall @devbazilio The unread message counts plugin only counts incoming messages if the user is online and receiving events. If you need to count unread messages when a user logs back, you will need to call chat.search() to retrieve missed messages and show counts on the application.

Note that ChatEngine also allows you to access the ChatEngine.pubnub object and send History & Batch History API calls.
https://www.pubnub.com/docs/web-javascript/api-reference-storage-and-playback#history

I will set this ticket to RESOLVED. If the issue persists, definitely feel free to open a ticket with PubNub Support (https://support.pubnub.com).
